runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -439,14 +440,14 @@ MM_CopyForwardScheme::updateLeafRegions(MM_EnvironmentVLHGC *env)
 				Assert_MM_true(updatedSpineRegion->containsObjects());
 
 				/* we need to move the leaf to another region's leaf list since its spine has moved */
-				region->_allocateData.removeFromArrayletLeafList();
+				region->_allocateData.removeFromArrayletLeafList(env);
 				region->_allocateData.addToArrayletLeafList(updatedSpineRegion);
 				region->_allocateData.setSpine((J9IndexableObject *)updatedSpineObject);
 			} else if (!isLiveObject(spineObject)) {
 				Assert_MM_true(isObjectInEvacuateMemory(spineObject));
 				/* the spine is in evacuate space so the arraylet is dead => recycle the leaf */
 				/* remove arraylet leaf from list */
-				region->_allocateData.removeFromArrayletLeafList();
+				region->_allocateData.removeFromArrayletLeafList(env);
 				/* recycle */
 				region->_allocateData.setSpine(NULL);
 				region->getSubSpace()->recycleRegion(env, region);
```

Comment:
consider removing this pass for offheap. basically keep a pool of ARRAYLET_RESERVED regions, and iterate only them to convert them to FREE if array surivived don't do anything (by not setting spine pointer in region descriptor during allocation)